rust/crd/src/lib.rs

```diff
@@ -230,6 +250,130 @@ impl HdfsCluster {
             .unwrap_or_default())
     }
 
+    /// Build the [`PersistentVolumeClaim`]s and [`ResourceRequirements`] for the given `rolegroup_ref`.
+    /// These can be defined at the role or rolegroup level and as usual, the
+    /// following precedence rules are implemented:
+    /// 1. group pvc
+    /// 2. role pvc
+    /// 3. a default PVC with 1Gi capacity
+    pub fn resources(
+        &self,
+        role: &HdfsRole,
+        rolegroup_ref: &RoleGroupRef<HdfsCluster>,
+    ) -> (Vec<PersistentVolumeClaim>, ResourceRequirements) {
+        let mut rg_resources = self.rolegroup_resources(role, rolegroup_ref);
+        let mut role_resources = self.role_resources(role, rolegroup_ref);
+        let mut default_resources = Some(self.default_resources(role, rolegroup_ref));
+
+        let r = [
+            default_resources.as_mut(),
+            role_resources.as_mut(),
+            rg_resources.as_mut(),
+        ]
+        .into_iter()
+        .flatten()
+        .reduce(|old, new| chainable_merge(new, old))
+        .unwrap(); // always succeeds because default_resources is Some()
+
+        let result_1 = r
+            .storage
+            .data
+            .build_pvc("data", Some(vec!["ReadWriteOnce"]));
+        let result_2 = r.clone().into();
+
+        (vec![result_1], result_2)
```

Comment:
I would not make this a vec just to fit the `volume_claim_templates`. Decreases testability etc.

Reply:
I don't see how this decreases testability.

Reply:
Ok let me rephrase then. Will there be more than one element in that vector in the future? If not, i dont see the point in returning a vec (simply to save on the `vec![xxx]`) when building the template?

Reply:
HDFS supports multiple volumes for it's data folders, so we might support that in the future. But to me, this is about encapsulation: I have one clear interface (this function) that returns PVCs and don't have to fiddle with them later in the code to pass them on to K8S.

Reply:
Ok did not know that. Then i resolve this.